uwblueprint/headsupguys
# Module experience user input

## Notion ticket link

[[Module Experience] User Input](https://www.notion.so/uwblueprintexecs/Module-Experience-User-Input-251594ce49dc4c97a5c1d55053e05eba)



## Implementation description
* Sets UserInput state when a user types in a module



## Steps to test
1. Create a module with at least 1 of each editable field (multiple choice, multiselect and short answer) and note its ID
2. While creating your module, ensure the module builder behaves as expected (still cannot edit fields in the module preview, can add new slides and buttons as usual)
3. Add a console.log(userInput) to line 32 of pages/module/[module]
![image](https://user-images.githubusercontent.com/24593519/158082166-9513dded-824d-4313-a82b-3ddf126713b0.png)
4. Navigate to http://localhost:3000/module/YOUR_MODULE_ID
5. Keep your browser's devtools console tab open as you edit the editable fields and take note of the changing userInput. If there are any errors, the schema doesn't behave as you expect, or you think I should do something differently (eg. delete items instead of setting them to false for multiselect) let me know!



## What should reviewers focus on?
* Module set state is working as intended


## Checklist
